test/scheduler_test.js

```diff
@@ -178,5 +182,22 @@ suite('Scheduler', function() {
       assume(new Date(updatedHook.lastFire.time) - new Date()).is.approximately(0, 2000); // 2s slop
       assume(updatedHook.nextScheduledDate).is.not.equal(oldScheduledDate);
     });
+
+    test('on error, notify is used with correct options', async () => {
+      creator.shouldFail = true;
+      hook.metadata.emailOnError = true;
+      await scheduler.handleHook(hook);
+
+      assume(scheduler.notify.credentials).exists();
+      assume(scheduler.notify.authorizedScopes).exists();
```

Comment:
wondering if we should check for credentials and authorizedScopes because I think tc-notify api doesn't expose any such thing. These checks might be misleading. What do you say ?